juju/juju
# Adds controller name to controller config.

### Checklist


----

## Description of change

Allows a user to now set a canonical name for a controller that will be passed through at bootstrap time.

## QA steps

```sh
juju bootstrap localhost batman # exec to the local lxd container and check /var/lib/juju/bootstrap-params, also check juju controller-config for controller-name set
juju bootstrap localhost  --config controller-name=batman1 # Will result in error saying controller name cannot be set via config
juju bootstrap localhost batman-arg --config controller-name=batman-config # should fail with an error
juju bootstrap localhost # controller name in juju controller-config should be defaulted to something like localhost-localhost
juju controller-config controller-name=batman2 # confirm changed with juju controller-config
```
